Lecture6(GithubAPI)/Component/BodyComponent.js

```diff
@@ -0,0 +1,84 @@
+import { useState, useEffect } from "react";
+import Skeleton from 'react-loading-skeleton'
+import 'react-loading-skeleton/dist/skeleton.css'
+import { SkeletonTheme } from "react-loading-skeleton";
+
+import SearchComponent from './SearchComponent';
+import CardContainerComponent from "./CardContainerComponent";
+
+
+const BodyComponent = () => {
+    const [filteredData, setFilteredData] = useState([]);
+    const [DataFromAPI, setDataFromAPI] = useState([]);
+    const [noRecord, setNoRecord] = useState(false);
+    const [loading, setLoading] = useState(true)
+    const githubUserNames = ["gavandivya", "akshaymarch7", "ap221882", "Bhallora", "aditifarkya", "it-abhishek2000", "deepak-kumar-dev", "soumyagangamwar", "shreerajjadeja", "nikitaj-57"]
+
+    const getGithubApiData = async () => {
+        try {
+            let githubdata = await Promise.all(githubUserNames.map(async (userName) => {
+                const UserInfo = await fetch(`https://api.github.com/users/${userName}`);
+                return await UserInfo.json();
+            }));
+            sessionStorage.setItem('teamData', JSON.stringify(githubdata));
+            return githubdata;
+        }
+        catch (error) {
+            console.log(error);
+        }
+    }
+
+    useEffect(() => {
+        if (!sessionStorage.getItem('teamData')) {
+            getGithubApiData().then((githubdata) => setDataFromAPI(githubdata))
+                .catch((err) => {
+                    console.log(err);
+                    setNoRecord(true);
+                })
+                .finally(() => setLoading(false));
+        }
+        else {
+            setDataFromAPI(JSON.parse(sessionStorage.getItem('teamData')));
```

Comment:
Very nice. Will add this to my code too. I was exhausting the API requests every time.